.github/workflows/test.yaml

```diff
@@ -0,0 +1,56 @@
+name: test
+
+on:
+  pull_request:
+
+jobs:
+  generate:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: true
+      - run: go get ./...
+      - run: go generate -x ./...
+      - uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.os }}-binaries
+          path: llm/llama.cpp/*/build/*/bin/*
+  lint:
+    needs: generate
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+          cache: false
+      - uses: actions/download-artifact@v4
+        with:
+          name: ubuntu-latest-binaries
+          path: llm/llama.cpp
+      - uses: golangci/golangci-lint-action@v3
+  test:
```

Comment:
is the uploading on downloading of artifacts much overhead? Im wondering if it would be more efficient to just do the `generate` and `test` as one job

Reply:
It's not and definitely faster than running generate in each job. v4 is also much faster according to their readme